mitodl/mit-open
# Ab/customize channel dropdown

### What are the relevant tickets?
closes https://github.com/mitodl/mit-open/issues/744

### Description (What does it do?)
This pr updates the list of dropdowns displayed in the Channel search based on the channel type

### How can this be tested?
Make some channels either through http://localhost:8063/api/v0/channels/ or the admin dashboard.  Make sure that field_search is set on your channels. Go to http://localhost:8063/c/<channel type>/channel name
Verify that the correct dropdowns are displayed for the channel type:

Topic: ["Department", "Level", "Resource Type"],
Department: ["Level", "Offered By", "Resource Type"],
Offeror:  ["Resource Type", "Platform", "Topic"],
Pathway: No dropdowns

Additionally, dropdowns should be hidden if the filter is used in field_search or if the search results return an empty aggregation array, which means that there are no items with values for the filter returned in the results. For example if you filter to 'offered_by=bootcamps' you should not see the `Department` dropdown since no bootcamps have departments
